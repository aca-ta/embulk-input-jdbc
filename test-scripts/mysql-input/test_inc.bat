mysql -uTEST_USER -pXXXXXXXX -DTESTDB < test.sql

DEL next.yml
del data\test000.00.csv
CALL embulk run test_inc.yml -c next.yml

echo "diff data/test_expected.csv data/test000.00.csv"
diff data/test_expected.csv data/test000.00.csv

IF "%ERRORLEVEL%" == "0" (ECHO "OK!") ELSE (ECHO "embulk-input-mysql inc(1) FAILED!")

mysql -uTEST_USER -pXXXXXXXX -DTESTDB < test_inc.sql
del data\test000.00.csv
CALL embulk run test_inc.yml -c next.yml

echo "diff data/test_inc_expected.csv data/test000.00.csv"
diff data/test_inc_expected.csv data/test000.00.csv

IF "%ERRORLEVEL%" == "0" (ECHO "OK!") ELSE (ECHO "embulk-input-mysql inc(2) FAILED!")
