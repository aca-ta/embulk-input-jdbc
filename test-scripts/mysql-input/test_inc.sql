INSERT INTO EMBULK_INPUT VALUES(
  4,
  44.4,
  NULL,
  NULL,
  NULL,
  NULL,
  NULL
);
INSERT INTO EMBULK_INPUT VALUES(
  5,
  55.5,
  NULL,
  NULL,
  NULL,
  NULL,
  NULL
);
