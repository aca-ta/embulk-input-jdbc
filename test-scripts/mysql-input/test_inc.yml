in:
  type: mysql
  host: localhost
  database: TESTDB
  user: TEST_USER
  password: XXXXXXXX
  table: EMBULK_INPUT
  select: "*"
  incremental: true
  incremental_columns: [ID]
  column_options:
    NUM: {type: string}
    DT: {type: string, timestamp_format: '%Y-%m-%d', timezone: "+0900"}
    DTTM0: {type: string, timestamp_format: '%Y-%m-%d %H:%M:%S', timezone: "+0900"}
    DTTM3: {type: string, timestamp_format: '%Y-%m-%d %H:%M:%S.%3N', timezone: "+0900"}
out:
  type: file
  path_prefix: data/test
  file_ext: csv
  formatter:
    type: csv
