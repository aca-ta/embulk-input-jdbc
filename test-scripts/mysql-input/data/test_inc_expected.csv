ID,NUM,STR,VARSTR,DT,DTTM0,DTTM3
4,44.40,,,,,
5,55.50,,,,,
